data-protocols/dsp/dsp-negotiation/dsp-negotiation-api/src/main/java/org/eclipse/edc/protocol/dsp/negotiation/api/controller/DspNegotiationController.java

```diff
@@ -123,13 +129,18 @@ public Map<String, Object> getNegotiation(@PathParam("id") String id, @HeaderPar
     public Map<String, Object> initiateNegotiation(@RequestBody(description = DSPACE_NEGOTIATION_CONTRACT_REQUEST_MESSAGE, required = true) JsonObject body, @HeaderParam(AUTHORIZATION) String token) {
         monitor.debug(() -> "DSP: Incoming ContractRequestMessage for initiating a contract negotiation.");
 
-        var negotiation = processMessage(token, Optional.empty(), body, DSPACE_NEGOTIATION_CONTRACT_REQUEST_MESSAGE, ContractRequestMessage.class, protocolService::notifyRequested);
+        var negotiation = processMessage(token,
+                Optional.empty(),
+                body,
+                DSPACE_NEGOTIATION_CONTRACT_REQUEST_MESSAGE,
+                ContractRequestMessage.class,
+                this::validateAndProcessRequest);
```

Comment:
note for the next milestone, refactor how we validate incoming requests because currently it's pretty scattered

Reply:
Added a marker issue #2936